packages/web/src/components/list/MultiDropdownList.js

```diff
@@ -386,15 +385,38 @@ class MultiDropdownList extends Component {
 		this.updateQueryOptions(this.props, true);
 	};
 
-	handleChange = (item) => {
+	handleChange = (e) => {
 		const { value, onChange } = this.props;
-		if (value === undefined) {
-			this.setValue(item);
-		} else if (onChange) {
-			onChange(item);
+		if (isEvent(e)) {
+			const { value: listValue } = e.target;
+			if (value === undefined) {
+				this.setValue(listValue);
+			} else if (onChange) {
+				onChange(listValue);
+			}
+		} else {
+			this.setValue(e);
```

Comment:
So I get the `isEvent` logic here is used to include user's integration for the click handling. I wonder what is the reason for adding `this.setValue(e);` here in the else block?

Reply:
I think if `e` passed in handleChange is value instead of an event, then it would be used for setting that value, but it should be `this.setValue(value)` right?

Reply:
Right, but `this.setValue(e)` is fine since we have to set the value which has been provided by user.

Reply:
This might change how controlled components operate.

Reply:
Not sure, but that's the same thing we're doing while rendering the list items inputs, on changing of which we call the `handleChange` function.
Exposing `handleChange` in render props provides the easiest way to update the values instead of managing own state.

Reply:
how does the handleChange work with the controlled component behavior? I think you're now bypassing it and directly updating the component's value ~ uncontrolled. How will it work in a controlled setting?

Reply:
I agree we should not violate common ground rules for input state management.

Reply:
If some user is using the controlled setting then there is no meaning to use `handleChange` method.
I just want to simplify the update process if someone is using a custom renderer so a user can easily update the current value without managing their own state.

Reply:
so we can't use custom-rendering and controlled behavior together?

Reply:
Yes, we can.
then you have to maintain state yourself and define your custom state changer function.

Reply:
I don't think exposing `handleChange` will break any React paradigm.
It's just for the ease of users so one can easily bind their inputs to the component's state.

Reply:
Okay. That makes sense. Thoughts @lakhansamani ?

Reply:
Don't we already have one such method for a user to manage state, i.e. `onChange` then why to expose `handleChange`?

Reply:
onChange won't get triggered here. It only works with controlled behavior - which is not handled via the handleChange method in the render prop here.

Reply:
Okay, then it makes sense, though this method can add more confusions to api.

Reply:
I agree. I think we should support the controlled behavior here as well - for consistency and ease of use. Here's how I'm thinking of it:

```js
handleChange = (e) => {
    let currentValue = e;
	if (isEvent(e)) { currentValue = e.target.value }

   // handle the behavior as earlier
}
```

Reply:
Yes, this looks good 👍

Reply:
Perfect, I'll do the changes.